psh/cp/cp.c

```diff
@@ -0,0 +1,185 @@
+/*
+ * Phoenix-RTOS
+ *
+ * cp - copy file
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <getopt.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+
+#include "../psh.h"
+
+#define SIZE_BUFF 256
+
+
+static void psh_cpinfo(void)
+{
+	printf("copy file");
+}
+
+
+static void psh_cp_help(const char *prog)
+{
+	printf("Usage: %s [options] SOURCE TARGET\n", prog);
+	printf("  -h:  shows this help message\n");
+}
+
+
+static int psh_cp(int argc, char **argv)
+{
+	FILE *src = NULL, *dst = NULL;
+	int fdsrc, fddst;
+	unsigned char buff[SIZE_BUFF];
+	char *filename, *destpath;
+	size_t destlen, filelen, cnt;
+	struct stat stat;
+	int c, retval = EXIT_SUCCESS, eexit = 0;
+
+	while ((c = getopt(argc, argv, "h")) != -1) {
+		switch (c) {
+			case 'h':
+				psh_cp_help(argv[0]);
+				eexit = 1;
+				break;
+			default:
+				psh_cp_help(argv[0]);
+				retval = EXIT_FAILURE;
+				eexit = 1;
+				break;
+		}
+	}
+
+	if (argc - optind != 2) {
+		psh_cp_help(argv[0]);
+		retval = EXIT_FAILURE;
+		eexit = 1;
+	}
+
+	while (eexit == 0) {
+		fdsrc = open(argv[optind], O_RDONLY);
+		if (fdsrc < 0) {
+			retval = EXIT_FAILURE;
+			fprintf(stderr, "cp: could not open source file\n");
```

Comment:
Pls take a look on our perror implementation :D I can use strerror, I guess

Reply:
IMHO `perror` is trivial to implement and should be the way to go in error path from libc call, I know implementing `perror` is not in the scope of this PR but I would want to userspace programs implementation in the right direction.

Reply:
note for the future: `perror("foo") == fprintf(stderr, "%s: %s", "foo", strerror(errno))` It would be good to maintain the same format (`: ` separator)

Reply:
Done, I've already use ":" for applet prefix, that's why I went with "-".